internal/web/server.go

```diff
@@ -38,60 +47,102 @@ type ServerConfigSupport struct {
 	BootstrapCheckError       error
 }
 
-func Start(ctx context.Context, support *ServerConfigSupport) error {
-	pkgTemplate, err := template.ParseFS(embededFs, "templates/packages.html")
+func (h *Hub) ws(w http.ResponseWriter, r *http.Request) {
+	wsClient, err := NewWsClient(h, w, r)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "Failed to create WebSocket client: %v", err)
+		return
 	}
-	supportTemplate, err := template.ParseFS(embededFs, "templates/support.html")
-	if err != nil {
-		return err
+	h.Register <- wsClient
+	go wsClient.HandleReads()
+	go wsClient.HandleWrites()
+}
+
+func init() {
+	loadTemplates()
+}
+
+func loadTemplates() {
+	templateFuncs := template.FuncMap{
+		"ToInstallButtonInput": components.ToInstallButtonInput,
 	}
+	baseTemplate = template.Must(
+		template.New("base.html").Funcs(templateFuncs).ParseFS(embededFs, "templates/layout/base.html"))
+	pkgsPageTmpl = template.Must(template.Must(baseTemplate.Clone()).
```

Comment:
I'm pretty sure this (templates location, parsing, nesting, splitting pages/components) needs more polishing, but I would continue it in #172, because there we have the case of adding another page + probably re-using the button again. And I think at this point it makes more sense to "craft" some more generic system than it is now.

Reply:
This might be a stupid question but why do we need different template instances for every page/component?

Reply:
To be honest I couldn't make it work any other way, but I can look into it again. What would you have expected instead? One `tmpl` instance that is used for everything?

Reply:
Yes, something like 
```
tpl := template.ParseFS(fs, "templates/pages/*.html", "templates/components/*.html", "...")
```

Reply:
because you still have to specify the "entrypoint" anyways…

Reply:
OK, let's keep it as is then.